spring-grpc-spring-boot-autoconfigure/src/test/java/org/springframework/grpc/autoconfigure/server/GrpcServerReflectionAutoConfigurationTests.java

```diff
@@ -14,45 +16,46 @@ public class GrpcServerReflectionAutoConfigurationTests {
 	private ApplicationContextRunner contextRunner() {
 		return new ApplicationContextRunner()
 			.withConfiguration(AutoConfigurations.of(GrpcServerReflectionAutoConfiguration.class))
-			.withBean("noopServerLifecylcle", GrpcServerLifecycle.class, Mockito::mock);
+			.withBean("noopServerLifecylcle", GrpcServerLifecycle.class, Mockito::mock)
+			.withBean(BindableService.class, Mockito::mock);
 	}
 
 	@Test
 	void whenReflectionEnabledFlagNotPresentThenCreateDefaultBean() {
-		this.contextRunner().run((context -> assertThat(context).hasSingleBean(BindableService.class)));
+		this.contextRunner().run((context -> assertThat(context).getBeanNames(BindableService.class).hasSize(2)));
 	}
 
 	@Test
 	void whenReflectionEnabledThenCreateBean() {
 		this.contextRunner()
 			.withPropertyValues("spring.grpc.server.reflection.enabled=true")
-			.run((context) -> assertThat(context).hasSingleBean(BindableService.class));
+			.run((context) -> assertThat(context).getBeanNames(BindableService.class).hasSize(2));
 	}
 
 	@Test
 	void whenReflectionDisabledThenSkipBeanCreation() {
 		this.contextRunner()
 			.withPropertyValues("spring.grpc.server.reflection.enabled=false")
-			.run((context) -> assertThat(context).doesNotHaveBean(BindableService.class));
+			.run((context) -> assertThat(context).getBeanNames(BindableService.class).hasSize(1));
 	}
 
 	@Test
 	void whenServerEnabledPropertySetFalseThenAutoConfigurationIsSkipped() {
 		this.contextRunner()
 			.withPropertyValues("spring.grpc.server.enabled=false")
-			.run((context) -> assertThat(context).doesNotHaveBean(BindableService.class));
+			.run((context) -> assertThat(context).getBeanNames(BindableService.class).hasSize(1));
 	}
 
 	@Test
 	void whenServerEnabledPropertyNotSetThenAutoConfigurationIsNotSkipped() {
-		this.contextRunner().run((context) -> assertThat(context).hasSingleBean(BindableService.class));
+		this.contextRunner().run((context) -> assertThat(context).getBeanNames(BindableService.class).hasSize(2));
 	}
 
 	@Test
 	void whenServerEnabledPropertySetTrueThenAutoConfigurationIsNotSkipped() {
 		this.contextRunner()
 			.withPropertyValues("spring.grpc.server.enabled=true")
-			.run((context) -> assertThat(context).hasSingleBean(BindableService.class));
+			.run((context) -> assertThat(context).getBeanNames(BindableService.class).hasSize(2));
```

Comment:
I think you will also want to add another test here as described in https://github.com/spring-projects/spring-grpc/pull/198/files#r2157685154